src/roles/crazed_shaman.py

```diff
@@ -0,0 +1,180 @@
+import re
+import random
+import itertools
+from collections import defaultdict, deque
+
+import botconfig
+from src.utilities import *
+from src import debuglog, errlog, plog, users, channels
+from src.functions import get_players, get_all_players, get_main_role, get_reveal_role, get_target
+from src.decorators import command, event_listener
+from src.containers import UserList, UserSet, UserDict, DefaultUserDict
+from src.dispatcher import MessageDispatcher
+from src.messages import messages
+from src.events import Event
+
+from src.roles._shaman_helper import havetotem, _get_target, _give_totem, _apply_totem
+
+TOTEMS = UserDict()         # type: Dict[users.User, str]
+LASTGIVEN = UserDict()      # type: Dict[users.User, users.User]
+SHAMANS = UserDict()        # type: Dict[users.User, List[users.User]]
```

Comment:
Prefer we add helper methods for this:
TOTEMS, LASTGIVEN, SHAMANS = setup_totem_vars()

The purpose of setup_totem_vars() is to create the three UserDicts and return them (as a tuple), but also to wire up events in the shared shaman code to do things like reset, begin_day, etc. Anywhere where the events are 100% the same across every shaman role (notably, transition_night_end is not included since the totem assignment and role messages code varies). This will reduce a lot of boilerplate.